packages/journey-manager/src/index.ts

```diff
@@ -211,17 +211,29 @@ export const run = (props: RunProps): RunOutput => {
 
   const triggeredSteps = getTriggeredSteps(props.config.conversationId);
 
-  // If all triggers have a URL constraint set and none match the current URL, fire a digression calback
+  /**
+   * Digression detection
+   * - if all triggers have a URL constraint set and none match the current URL, fire a digression callback
+   * - notable exception: if the step is used as escalation or end as well, missing URL constraint is ok.
```

Comment:
rewriting to confirm I understand what you're saying
```suggestion
   * - notable exception: ignore URL constraints if if the step is used as an escalation or end.
```